src/script/api/script_object.cpp

```diff
@@ -312,7 +312,7 @@ ScriptObject::ActiveInstance::~ActiveInstance()
 	if (!StrEmpty(text) && (GetCommandFlags(cmd) & CMD_STR_CTRL) == 0) {
 		/* The string must be valid, i.e. not contain special codes. Since some
 		 * can be made with GSText, make sure the control codes are removed. */
-		::str_validate(const_cast<char *>(text), text + strlen(text), SVS_NONE);
+		::StrMakeValidInPlace(const_cast<char *>(text), SVS_NONE);
```

Comment:
The `const_cast<char *>` is not needed anymore, since the `StrMakeValidInPlace` without `last`'s `str` parameter is `const char *`.